core-plugins/src/main/java/co/cask/hydrator/plugin/batch/file/AbstractMetadataInputFormat.java

```diff
@@ -118,54 +122,37 @@ public RecordReader createRecordReader(InputSplit inputSplit, TaskAttemptContext
     return recordReader;
   }
 
-  private void recursivelyAddFileStatus (List<FileStatus> fileStatuses, List<String> baseFolders,
+  private void recursivelyAddFileStatus(List<AbstractFileMetadata> fileMetadataList,
```

Comment:
lets add a comment saying this is used instead of FileSystem.listFiles() because that method doesn't list empty directories.